apps/publish.go

```diff
@@ -545,3 +547,31 @@ func getLocalFileData(link string) ([]byte, error) {
 
 	return ioutil.ReadFile(link)
 }
+
+func remoteFileDataExists(link string) (bool, error) {
+	_, err := getRemoteFileData(link)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func localFilePathExists(link string) (bool, error) {
+
+	logoFileInfo, err :=  os.Stat(link)
+	if err != nil {
+		return false, err
+	} else if logoFileInfo.IsDir() {
```

Comment:
don't need this else case here because will return in the if statement above

Reply:
We actually need to explicitly check that the file path isn't a directory, because os.Stat won't produce an error if it is, but it will throw an error when we actually go to read the file in `Publish` via `getFileData`. You can check that the app will pass the validate step if you remove lines 565-567 from `apps/publish.go` and  ketch-manifest.yml passes a directory (that exists) as the logo link, but the publish step will crash.